src/adapter/http/src/middleware/auth_policy_layer.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use axum::body::Body;
+use axum::response::Response;
+use http::StatusCode;
+use kamu_accounts_services::AuthPolicyService;
+use tower::{Layer, Service};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct AuthPolicyLayer {}
+
+impl AuthPolicyLayer {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl<Svc> Layer<Svc> for AuthPolicyLayer {
+    type Service = AuthPolicyMiddleware<Svc>;
+
+    fn layer(&self, inner: Svc) -> Self::Service {
```

Comment:
How is this one different from FlightSQL? Can't we just have a common single Axum middleware?